hibernate-core/src/main/java/org/hibernate/collection/internal/AbstractPersistentCollection.java

```diff
@@ -647,25 +654,22 @@ public final boolean setCurrentSession(SharedSessionContractImplementor session)
 		if ( session == this.session ) {
 			return false;
 		}
-		else {
-			if ( this.session != null ) {
-				final String msg = generateUnexpectedSessionStateMessage( session );
-				if ( isConnectedToSession() ) {
-					throw new HibernateException(
-							"Illegal attempt to associate a collection with two open sessions. " + msg
-					);
-				}
-				else {
-					LOG.logUnexpectedSessionInCollectionNotConnected( msg );
-					this.session = session;
-					return true;
-				}
+		else if ( this.session != null ) {
+			final String msg = generateUnexpectedSessionStateMessage( session );
+			if ( isConnectedToSession() ) {
+				throw new HibernateException(
+						"Illegal attempt to associate a collection with two open sessions. " + msg
+				);
 			}
 			else {
-				this.session = session;
-				return true;
+				LOG.logUnexpectedSessionInCollectionNotConnected( msg );
 			}
 		}
+		if ( hasQueuedOperations() ) {
+			LOG.queuedOperationWhenAttachToSession( MessageHelper.collectionInfoString( getRole(), getKey() ) );
+		}
```

Comment:
Logs a warning when a collection with queued operations is attached to the session.

This is known to happen when calling `Session#save`, `#saveOrUpdate`, or `#update` on an entity with a collection with queued operations. Provided the collection is not the target of a merge operation, the queued operations will be executed and cleared when the collection is initialized or the session is flushed.